packages/babel-core/src/transformation/file/index.js

```diff
@@ -410,8 +412,32 @@ export default class File extends Store {
   }
 
   parse(code: string) {
+    let opts = this.opts;
+    let parseCode = parse;
+    if (opts.parserOpts.parser) {
+      parseCode = opts.parserOpts.parser;
+
+      if (typeof parseCode === "string") {
+        let dirname = opts.parserOpts.dirname || process.cwd();
+        let parser = resolve(parseCode, dirname);
+        if (parser) {
+          parseCode = require(parser).parse;
+        } else {
+          throw new Error(`Couldn't find parser ${parseCode} with "parse" method relative to directory ${dirname}`);
+        }
+      }
+
+      if (!this.parserOpts.parser) {
+        this.parserOpts.parser = {
```

Comment:
Ok so this is setting the default parser for `recast` to babylon 6 because recast uses esprima by default?